### PR TITLE
feat(provisioning): add partnership:restricted flag to OrganizationMember model

### PR DESCRIPTION
**Pull request by cathteng**
For partnerships we want to create org managers that cannot be removed by the owner. This can be done by adding a flag on the `OrganizationMember` model and checking for it in APIs and on the frontend, much like how the `idp:provisioned` flag is used today.

For ER-1962

### CODE REVIEW

**Review by schew2381 — COMMENTED** [review]
Code change lgtm, will wait for migration approval

**Review by markstory — APPROVED** [review]
Migration is good. (no ddl changes)

**Pull request merged**
